(add) New project (#140)
Codesee-io/oss-port: projects/Gautam-Arora24/resize-observer-hook.mdx

---

> name: resize-observer-hook.

Reviewer: ```suggestion
name: resize-observer-hook
```

repoUrl: http://github.com/Gautam-Arora24/resize-observer-hook
description: ⚛️ A React Hook to monitor changes in the size of an element using native ResizeObserver API 🔍
languages: # A list of programming languages used in this project
  - JavaScript
  - HTML / CSS
  - React
tags:
  - First Timer Friendly

>   - Resize-Observer

Reviewer: ```suggestion
```

avatar: avatar.png
websiteUrl: https://gautam-arora24.github.io/resize-observer-hook/
twitterUrl: http://twitter.com/Gautamarora24
currentlySeeking:
  - Developers
  - Frontend Developers
  - Maintainers
  - Code Reviewers
  - Mentors
  - Testers
contributionOverview:
  mainLocation: India
  isMentorshipAvailable: true
learnLinks: #
  - title: ResizeObserver API
    url: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry
---

<!-- Divide your content into sections using our handy shortcodes. Each section is optional -->

<Overview>
"resize-observer-hook" is a ⚛️ React hook that monitors an element (provided with ref) and provides the height and width of that element dynamically using Native Resize Observer API.
<br/>
<br/>

> We usually use media queries to change the style of a particular element but media queries are great for updating layouts at specific points when the viewport changes sizes, but what if we want to change the style of an element based on the size of another element. We can use this library for the same case.

Reviewer: ```suggestion
We usually use media queries to change the style of a particular element as media queries are great for updating layouts at specific points when the viewport changes sizes. But what if we want to change the style of an element based on the size of another element? In that case, we can use this library!
```

> Currently, the library is in the initial stage, and there are a lot more things that need to be added like Types support, Testing for different cases, more props and etc. So, if you are interested to contribute to the library, you are welcomed 🙂.

Reviewer: ```suggestion
Currently, the library is in the initial stage, and there are a lot more things that need to be added like Types support, testing for different cases, more props, etc. So, if you are interested to contribute to the library, you are welcomed 🙂.
```

<br/>Do ⭐️ the repo.
<br/>
Any feedback or suggestions are appreciated 👋

</Overview>

<Contributing>

> - If you feel like there is some kind of bug or any other issue that is concerning for the library. Feel free to create an issue. I will try my best to solve it as soon as possible.

Reviewer: ```suggestion
- If you feel that there may be some kind of bug or any other issue that is concerning for the library, feel free to create an issue. I will try my best to solve it as soon as possible.
```
